File: daily_motion_player_flutter_interface/ios/Classes/DailyMotionPlayerFlutterInterfacePlugin.swift
import Flutter
import UIKit
import DailymotionPlayerSDK

public class DailyMotionPlayerFlutterInterfacePlugin: NSObject, FlutterPlugin {
    public static func register(with registrar: FlutterPluginRegistrar) {
        let factory = DailymotionPlayerViewFactory(messenger: registrar.messenger())
        registrar.register(factory, withId: "dailymotion-player-view")
    }
}

File: daily_motion_player_flutter_interface/android/src/main/kotlin/com/lol/daily_motion_player_flutter_interface/DailyMotionPlayerFlutterInterfacePlugin.kt
package com.lol.daily_motion_player_flutter_interface

import io.flutter.embedding.engine.plugins.FlutterPlugin

/** DailyMotionPlayerFlutterInterfacePlugin */
class DailyMotionPlayerFlutterInterfacePlugin : FlutterPlugin {

  override fun onAttachedToEngine(binding: FlutterPlugin.FlutterPluginBinding) {
    binding.platformViewRegistry.registerViewFactory(
      "dailymotion-player-view",
      DailymotionPlayerViewFactory(
        messenger = binding.binaryMessenger,
        context = binding.applicationContext
      )
    )
  }

  override fun onDetachedFromEngine(binding: FlutterPlugin.FlutterPluginBinding) {}
}
